### PR TITLE
passes cache config to job

### PR DESCRIPTION
**Pull request by ankitaluthra1**
### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: internal/cache/file/downloader/job.go [hunk 1]
```diff
@@ -60,7 +61,6 @@ type Job struct {
 	fileSpec             data.FileSpec
 	// Specifies whether the CRC check needs to be done after the file is
```
Comment (sethiay) [inline internal/cache/file/downloader/job.go:62]: remove the comment of field being removed.

File: internal/cache/file/downloader/job.go [hunk 2]
```diff
@@ -87,7 +87,8 @@ type Job struct {
 	// is responsibility of JobManager to pass this function.
 	removeJobCallback func()
 
-	mu locker.Locker
+	mu              locker.Locker
+	fileCacheConfig *config.FileCacheConfig
```
Comment (sethiay) [inline internal/cache/file/downloader/job.go:91]: move the fileCacheConfig just below fileSpec and in place of enableCrcCheck

File: internal/cache/file/downloader/job.go (hunk 1 above)
Comment (ankitaluthra1) [reply internal/cache/file/downloader/job.go]: done

File: internal/cache/file/downloader/job.go (hunk 2 above)
Comment (ankitaluthra1) [reply internal/cache/file/downloader/job.go]: done

**Review by sethiay — APPROVED** [review]

**Pull request merged**